### DISCUSSION TITLE
replaygain: apply to multi-discs albums (CD-box set)

### DISCUSSION THREAD

**Discussion by ghost** [quodlibet/quodlibet#4081] [General]
Hi, 
having in mind [this discussion](https://hydrogenaud.io/index.php?topic=112389.0) about how to apply replay gain to each disc of a CD Box set with foobar, I was wondering whether there is some way to do the same with Quodlibet. In Foobar you can set the preference on how to scan the files (e.g.:  %album artist% | %date% | %album% | %discnumber%.) which will make Foobar scanning each disc of box set as a single album. I couldn't find a similar setting in QL. I would be grateful if someone could help.
If there is no possibility is there a way to "export" so to say a set of files to foobar so that it can do it? 
I have a huge number of boxset and I would like to apply Replay gain on per disc basis. 

Best 
Giusuin

**Comment by declension** [comment]
I've had a look, unfortunately this is hard.

The plugin is registered to handle albums, so is executed via `plugin_albums()`. This then means it is already _handed_ albums and these are defined by core code to be based on `album_key` which is based on _one_ of tags `album_grouping_key`, `labelid or `musicbrainz_albumid`, _and_ `albumsort` _and_ `albumartistsort`.

Note how `discnumber` is _not_ present (and changing this would bork loads of things in loads of views, so albums will always be all discs here (unless you hack one of those tags - that's one workaround)

The plugin _could_ be updated to not work with albums but that would be negative for most users. Or maybe there could be a way of specifying any plugin to run as songs only (not albums), and then custom groupings in this plugin somehow, but again... hard... and risky 

> If there is no possibility is there a way to "export" so to say a set of files to foobar so that it can do it?

To export, yes, that's one way. Either playlists (with XSPF - but you'll need the latest unreleased version I suspect), or use the _Custom Commands_ plugin and some hackery

**Reply by ghost** [reply]
I see, thank you for your answer; i tries exporting the playlist of files either as a m3u or a pls list; but foobar doesn’t recognise it..
Thank you anyway.
Best
